src/lattice_reaction_system_diffusion.jl

```diff
@@ -0,0 +1,501 @@
+    only_diff = [(s in diff_species) && !Base.isstored(ns_jac_prototype, s, s)
+                 for s in 1:(lrs.nS)]
+
+    # Declares sparse array content.
+    J_colptr = fill(1, lrs.nC * lrs.nS + 1)
+    J_nzval = fill(0.0,
+                   lrs.nC * (nnz(ns_jac_prototype) + count(only_diff)) +
+                   length(edges(lrs.lattice)) * length(diffusion_rates))
+    J_rowval = fill(0, length(J_nzval))
+
+    # Finds filled elements.
+    for comp in 1:(lrs.nC), s in 1:(lrs.nS)
+        col_idx = get_index(comp, s, lrs.nS)
+
+        # Column values.
+        local_elements = in(s, diff_species) *
+                         (length(lrs.lattice.fadjlist[comp]) + only_diff[s])
+        diffusion_elements = -(ns_jac_prototype.colptr[(s + 1):-1:s]...)
+        J_colptr[col_idx + 1] = J_colptr[col_idx] + local_elements + diffusion_elements
+
+        # Row values.
+        rows = ns_jac_prototype.rowval[ns_jac_prototype.colptr[s]:(ns_jac_prototype.colptr[s + 1] - 1)] .+
+               (comp - 1) * lrs.nS
+        if in(s, diff_species)
+            # Finds the location of the diffusion elements, and inserts the elements from the non-spatial part into this.
+            diffusion_rows = (lrs.lattice.fadjlist[comp] .- 1) .* lrs.nS .+ s
+            split_idx = isempty(rows) ? 1 : findfirst(diffusion_rows .> rows[1])
+            isnothing(split_idx) && (split_idx = length(diffusion_rows) + 1)
+            rows = vcat(diffusion_rows[1:(split_idx - 1)], rows,
+                        diffusion_rows[split_idx:end])
+            if only_diff[s]
+                split_idx = findfirst(rows .> get_index(comp, s, lrs.nS))
+                isnothing(split_idx) && (split_idx = length(rows) + 1)
+                insert!(rows, split_idx, get_index(comp, s, lrs.nS))
+            end
+        end
+        J_rowval[J_colptr[col_idx]:(J_colptr[col_idx + 1] - 1)] = rows
+    end
+
+    # Set element values.
+    if !set_nonzero
+        J_nzval .= 1.0
+    else
+        for (s_idx, (s, rates)) in enumerate(diffusion_rates),
+            (e_idx, edge) in enumerate(edges(lrs.lattice))
+
+            col_start = J_colptr[get_index(edge.src, s, lrs.nS)]
+            col_end = J_colptr[get_index(edge.src, s, lrs.nS) + 1] - 1
+            column_view = @view J_rowval[col_start:col_end]
+
+            # Updates the source value.
+            val_idx_src = col_start +
+                          findfirst(column_view .== get_index(edge.src, s, lrs.nS)) - 1
+            J_nzval[val_idx_src] -= get_component_value(rates, e_idx)
+
+            # Updates the destination value.
+            val_idx_dst = col_start +
+                          findfirst(column_view .== get_index(edge.dst, s, lrs.nS)) - 1
+            J_nzval[val_idx_dst] += get_component_value(rates, e_idx)
+        end
+    end
+
+    return SparseMatrixCSC(lrs.nS * lrs.nC, lrs.nS * lrs.nC, J_colptr, J_rowval, J_nzval)
+end
+
+### JumpProblem ###
+
```

Comment:
It feels like the `JumpProblem` codebloack below belongs in its own separate file. As well as the ODE codebloack above. The reason is that they do not interact. It leads to better maintanability. For a simple example, when you update the `JumpProcesses.JumpProblem` in subsequent PRs, it will be nicer to have that change made in a file that is 70 lines long.

You might also want to make a `src/spatial` folder and put the new files there.

Reply:
See my comment below for a suggestion for how to split it. Just a suggestion though.

Reply:
done